geode-core/src/main/java/org/apache/geode/internal/cache/xmlcache/CacheXmlParser.java

```diff
@@ -2596,6 +2596,18 @@ private void endDeclarable() {
 
   public void startElement(String namespaceURI, String localName, String qName, Attributes atts)
       throws SAXException {
+    // This while loop pops all StringBuffers at the top of the stack
+    // that contain only whitespace; see GEODE-3306
+    while (!stack.empty()) {
+      Object o = stack.peek();
+      if (o instanceof StringBuffer
+          && ((StringBuffer) o).toString().replaceAll("\\s", "").equals("")) {
```

Comment:
I think `StringUtils.isBlank( (StringBuffer o).toString())` might be simpler here as well as in `endElement`.